### PR TITLE
DateRangePicker widget improvements

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Improving the date range picker with the following functionalities;

1. Remove milliseconds granularity (Issue  #988).
2. Allow user to configure available granularities (Issue #994).
3. Display high/low duration pickers based on available granularities (Issue #994).
4. Auto-sync on user defined intervals (Issue #994).
5. Highlight selected time range (Issue #994)
6. Load default time range based on available granularities
7. Provide a popover for custom range selection. 
8. Load from date range details provided as query param.
9. Display selected date range details as query param.
10. Change date format from DD-MMM-YYYY to YYYY-MM-DD.

**NOTE: Please review do not merge**

### CODE REVIEW

File: samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx [hunk 1]
```diff
@@ -17,97 +17,236 @@
  */
 
 import React from 'react';
-import { Button, IconButton } from 'material-ui';
-import HighGranularityMode from '@material-ui/icons/KeyboardArrowRight';
-import LowGranularityMode from '@material-ui/icons/KeyboardArrowLeft';
+import {IconButton, FlatButton, Popover, Menu} from 'material-ui';
+import {HardwareKeyboardArrowRight, HardwareKeyboardArrowLeft} from 'material-ui/svg-icons';
+import CustomTimeRangeSelector from "./CustomTimeRangeSelector";
```
Comment (Niveathika92) [inline samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx]: Use single quotes

**Comment by Niveathika92** [comment]
Let's hold this ATM and not add for the SP4.3.0-alpha release.

File: samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx [hunk 2]
```diff
@@ -17,97 +17,236 @@
...
+    getSelectedGranularityLevel() {
+        let selectedTimeRange = this.getSelectedTimeRange();
+        if (this.getHighGranularityOptions().indexOf(selectedTimeRange) > -1) {
+            return 'high';
+        } else if (this.getLowGranularityOptions().indexOf(selectedTimeRange) > -1) {
+            return 'low';
+        } else if (this.getHighGranularityOptions().indexOf(this.props.options.defaultValue) > -1) {
+            return 'high';
+        } else if (this.getLowGranularityOptions().indexOf(this.props.options.defaultValue) > -1) {
+            return 'low';
```
[the remark this answers is not in the document]
Comment (ruwiniwj) [reply samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx]: If we simplify this as ||, in a situation where the URL query params specify a time range in low granularity options but the default value selected is in high granularity options, the date range picker will load the details for the selected time range but display the wrong granularity options on the top. This logic written without || to give priority to the selected time range in query param over the default value selected.

File: samples/widgets/DateTimeRangePicker/package.json [hunk 3]
```diff
@@ -3,10 +3,9 @@
     "version": "0.1.0",
     "private": true,
     "dependencies": {
-        "@material-ui/icons": "1.0.0-beta.43",
-        "@wso2-dashboards/widget": "^1.2.0",
+        "@wso2-dashboards/widget": "^1.2.2",
         "lodash": "^4.17.10",
-        "material-ui": "1.0.0-beta.45",
+        "material-ui": "^0.20.0",
```
[the remark this answers is not in the document]
Comment (ruwiniwj) [reply samples/widgets/DateTimeRangePicker/package.json]: The custom time range configuration panel needed to be displayed as a popover. Neither the popover or dialog box in material ui 1.0.0 got rendered properly, so we downgraded to material ui 0.20.0 as the widget doesn't use anything that is not supported in material ui 0.20.0.

**Comment by TanyaM** [comment]
Closing this PR since this feature is moved to analytics solutions repo.

File: samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ruwiniwj) [reply samples/widgets/DateTimeRangePicker/src/GranularityModeSelector.jsx]: Yes, we give priority to time range specified in query params over the default time range selected.

**Pull request closed without merging**